### DISCUSSION TITLE
Rule to update docker images to be prefixed with private registry url

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#19395] [Request Help] [answered]
Hey all,

Asking here to see if this situation is possible using the current functionality, or whether this would be a feature request.

I would like to be able to configure Renovate, so that it will raise PRs to update any Docker image statement (Dockerfile, bitbucket-pipelines, etc) so that it will use our private docker registry and not the public Docker Hub one.

For example:  Update something like `FROM node:18` to be `FROM some.private.mirror/node:18`

Perhaps this could be thought of as a variation on a Docker manager replacement rule (#14149)

**Comment by rarkins** [comment]
I was sure someone contributed a regex manager trick really recently to use autoReplaceTemplate to achieve the same, but I can't find it

**Reply by setchy** [reply]
Thanks @rarkins - this was enough to point me in the right direction.

Found an example in the docs under [autoReplaceStringTemplate](https://docs.renovatebot.com/configuration-options/#autoreplacestringtemplate).

This should be enough to get us going, though it would be nice to have a simpler way to set a docker private registry in the renovate configuration and internalize all the replacement logic.  What are your thoughts about that?

**Reply by setchy** [reply]
So I've tried a few different ways to achieve this, but have had very limited success with prefixing all of our non-private registried image statements...

Appreciate any further guidance/ tips this community might have...

Also interested in feedback if a new configuration property would be worthwhile contributing, which should provide a more explicit / simple method than package rules with autoReplaceStringTemplate regexs or replacement rules

**Reply by setchy** [reply]
Feature enhancement #20905 raised to improve DX for configuring these types of migrations / replacements

**Comment by setchy** [comment, marked as answer]
This is now possible via `replacementNameTemplate` from [v35.16.0](https://github.com/renovatebot/renovate/releases/tag/35.16.0) onwards
